blockprod/src/detail/tests.rs

```diff
@@ -851,16 +924,23 @@ mod produce_block {
                 mock_job_manager.expect_add_job().times(1).returning(move |_, _| {
                     let (_, cancel_receiver) = unbounded_channel::<()>();
                     let mut rng = make_seedable_rng(seed);
-                    let job_key =
-                        JobKey::new(None, Id::new(H256::random_using(&mut rng)) as Id<GenBlock>);
-                    Ok((job_key, cancel_receiver))
+                    let job_key = JobKey::new(
+                        vec![],
+                        Id::new(H256::random_using(&mut rng)) as Id<GenBlock>,
```

Comment:
Interesting construct with the `as` conversion. If I understand this correctly, I think using `Id::<GenBlock>::new(...)` would be more idiomatic and also easier to understand. NO conversion is actually happening despite `as` suggesting so. Did I miss something? This is replicated a number of times.

Reply:
Yep you're right - there's no casting done here, I just prefer it out of the way... but yeah, fixing because it's not idiomatic.